runtime/sema/check_composite_declaration.go

```diff
@@ -2052,6 +2053,44 @@ func (checker *Checker) checkDefaultDestroyParamExpressionKind(
 	}
 }
 
+func (checker *Checker) checkDefaultDestroyEventParam(
+	param Parameter,
+	index int,
+	constructorFunctionParameters []*ast.Parameter,
+	containerType ContainerType,
+	containerDeclaration ast.Declaration,
+) {
+	paramType := param.TypeAnnotation.Type
+	paramExpr := constructorFunctionParameters[index]
```

Comment:
Instead of passing all constructor function parameters and the index, pass the `ast.Parameter` from the call-site directly. Maybe also rename `paramExpr`, as it is not an expression.